src/yadism/coefficient_functions/coupling_constants.py

```diff
@@ -92,9 +110,16 @@ def leptonic_coupling(self, mode, quark_coupling_type):
         # load Z coupling
         projectile_v = 0.0
         projectile_a = 0.0
-        if mode in ["phZ", "ZZ"]:
+        projectile_BSM_v = 0.0
+        projectile_BSM_a = 0.0
+        if mode in ["phZ", "ZZ", "Z4F"]:
             projectile_v = self.vectorial_coupling(abs(projectile_pid))
             projectile_a = self.weak_isospin_3[abs(projectile_pid)]
+            projectile_BSM_v = self.BSM_coupling_vectorial[abs(projectile_pid)]
```

Comment:
Maybe for consistency you should also rename `projectile_v` to `projectile_Z_v` and same for `_a`. 
Or maybe this has to be even generalized more if you have multiple operators...

Reply:
Hi Giacomo,
This makes good sense, thanks! I have implemented it